src/sync-pending-events.ts

```diff
@@ -255,4 +258,34 @@ async function getLatestBlockNumber(provider: string) {
     }
   });
   return latestBlockNumber
+}
+
+async function getValidBlockNumber(provider: string, startPoint: number, endPoint: number) {
```

Comment:
1. if (pastEvents.breakpoint > lastCheckedBlockNumber), what we should return. (null)
2. if success, return endPoint (success meaqns endPoint = inputEndPoint
3. If some issue happens, like some blockNumber does not have blockHeader (not existing), need console log them.